futures-util/src/async_await/select.rs

```diff
@@ -1,5 +1,30 @@
-//! The `select` macro.
-
+/// The `select` macro.
+/// 
+/// This macro polls multiple futures simultaneously, executing the branch that finishes first.
+/// The output of the future shadows the name of the future within each branch.
```

Comment:
This one's tricky... maybe: "Inside each branch, the respective future's output is available via a variable with the same name as the future."